vj4/handler/contest.py

```diff
@@ -86,12 +87,14 @@ class ContestCommonOperationMixin(object):
     tdoc, tsdocs = await contest.get_and_list_status(self.domain_id, tid)
     if not self.can_show_scoreboard(tdoc):
       raise error.ContestScoreboardHiddenError(self.domain_id, tid)
-    udict, pdict = await asyncio.gather(user.get_dict([tsdoc['uid'] for tsdoc in tsdocs]),
-                                        problem.get_dict(self.domain_id, tdoc['pids']))
+    udict, dudict, pdict = await asyncio.gather(
+        user.get_dict([tsdoc['uid'] for tsdoc in tsdocs]),
+        domain.get_dict_user_by_uid(self.domain_id, [tsdoc['uid'] for tsdoc in tsdocs]),
+        problem.get_dict(self.domain_id, tdoc['pids']))
     ranked_tsdocs = contest.RULES[tdoc['rule']].rank_func(tsdocs)
     rows = contest.RULES[tdoc['rule']].scoreboard_func(is_export, self.translate, tdoc,
-                                                       ranked_tsdocs, udict, pdict)
-    return tdoc, rows
+                                                       ranked_tsdocs, udict, pdict, dudict=dudict)
```

Comment:
you can put `dudict` in between `udict` and `pdict` since scoreboard_func is an internal interface.